ts/src/connections/sshHelpers.ts

```diff
@@ -2,12 +2,33 @@
 // Licensed under the MIT license.
 
 import * as ssh from '@vs/vs-ssh';
+import { IncomingMessage } from 'http';
 import {
     client as WebSocketClient,
     connection as WebSocketConnection,
     IClientConfig,
 } from 'websocket';
 
+declare module 'websocket' {
+    // eslint-disable-next-line @typescript-eslint/tslint/config
+    interface client {
+        /**
+         * 'httpResponse' event in WebSocketClient is fired when the server responds but the HTTP request doesn't properly upgrade to a web socket,
+         * i.e. the status code is not 101 `Switching Protocols`. The argument of the event callback is the recieved response.
+         */
+        on(event: 'httpResponse', cb: (response: IncomingMessage) => void): this;
+    }
+}
+
+/**
+ * Error class for errors connecting to a web socket in non-node (browser) context.
+ * There is no status code or underlying network error info in the browser context.
```

Comment:
I wonder if the relay service should do something different for rejecting browser websocket connection attempts, in order to propagate error info. Maybe it could accept the websocket and then immediately close it with a websocket status code.

Not blocking this PR, but maybe worth considering later.

Reply:
I see the link below to https://websockets.spec.whatwg.org/#eventdef-websocket-error gives some reasoning for not supplying any error details, but I'm not convinced that reasoning applies to our scenario.

Reply:
The error may happen upstream to our service, and it won't have a chance to respond.
That's really up to the user agent (browser) as to what data is provided in the web socket error, not sure if we can control that from our SDK.
Anyway, `RelayTunnelConnector` treats this error differently and retries several times before giving up.

Reply:
> The error may happen upstream to our service

In some cases, yes, but for some errors (such as tunnel not found or host not connected) the relay service could return an error code in the websocket close message (which is exposed by browsers: https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/code) or could even send details over the websocket stream.